### ISSUE TITLE
efficientnet: any plans to pretrain using group norm?

### ISSUE THREAD

**Issue by keven425** [tensorflow/tpu#681]
Currently efficient net uses batch normalization layers, which makes training dependent on batch size. It would be nice to have pretrained models using group normalization.

**Comment by keven425** [comment]
Any thoughts?

**Comment by mingxingtan** [comment]
Do you mind prepare a PR for replacing BN with GN? After your PR is ready, I will try to retrain a model.

**Comment by keven425** [comment]
@mingxingtan I'm looking into the code base. Currently, in [efficientnet_model.py](https://github.com/tensorflow/tpu/blob/master/models/official/efficientnet/efficientnet_model.py), it uses `bn` as variable names extensively, and `[GroupNormalization](https://www.tensorflow.org/addons/api_docs/python/tfa/layers/GroupNormalization)` requires an additional `groups` parameter, I'm deciding between these two approaches:

1. rename the variable names which contains `bn` to `gn`, add `groups` parameter to constructor of normalization layers. i.e. make the code base work exclusively with group norm.
2. keep the variable name `bn`. Refactor the code to support group norm, in additional to batch norm. Pass in `groups` parameter via keyword arg into constructor.

Any recommendations?

**Comment by mingxingtan** [comment]
@keven425 Would be great if you could keep both BN and GN for now.  If GN is indeed much better, then we can delete the BN code in future.

Do you mind I assign this issue to you if you are looking into this issue? Thanks!

**Comment by SergioG-M** [comment]
Any update on this?

**Comment by Riddhasree** [comment]
Can anyone please tell how do we choose the group size for group normalization?
